acquia_lift.drush.inc

```diff
@@ -32,3 +36,43 @@ function drush_acquia_lift_menu_rebuild() {
   _acquia_lift_build_menu('acquia-lift-controls');
   drush_log('The menu has been rebuilt.', 'success');
 }
+
+/**
+ * Syncs the Acquia Lift Campaigns.
+ *
+ * This command is an alternative to the UI's "Sync with Lift" buttons.
+ */
+function drush_acquia_lift_campaign_sync() {
+  module_load_include('inc', 'acquia_lift', 'acquia_lift.batch');
+
+  // Step 1. Select sync type.
+  $options = array(
+    'all' => dt('Sync all campaigns.'),
+    'one' => dt('Sync one campaign by its machine name.')
+  );
+  $type_selection = drush_choice($options, dt('Select Sync Type'));
+  if (!$type_selection) {
+    return;
+  }
+  $agents = array();
+  foreach (acquia_lift_get_agent_types() as $type => $info) {
+    $agents = array_merge($agents, personalize_agent_load_by_type($type));
+  }
+
+  // Step 2a. Select a campaign (if the type is not "all").
+  if ($type_selection == 'one') {
+    $agent_names = array_keys($agents);
+    $name_selection = drush_choice($agent_names, dt('Select A Campaign Machine Name to Sync'));
+    if (!$name_selection) {
+      return;
+    }
+    $agent_name = $agent_names[$name_selection];
+    $agent = $agents[$agent_name];
+    $agents = array($agent_name => $agent);
+  }
+
+  // Step 3. Sync selected campaign(s)
+  acquia_lift_batch_sync_campaigns($agents);
```

Comment:
I think it is no difficulty to do as requested. The reason I didn't go for that route was because we would then not be able to reuse acquia_lift_batch_sync_campaigns(), and have to maintain two sets of code with almost the same purpose.

Let me try refactoring it using native batch.